metadata_backend/api/middlewares.py

```diff
@@ -0,0 +1,50 @@
+"""Middleware methods for server."""
+import json
+from typing import Callable
+
+from aiohttp.web import HTTPError, Request, Response, middleware
+
+
+def error_middleware() -> Callable:
+    """Middleware for handling exceptions recieved from the API methods.
+
+    :returns: Handled response
+    """
+
+    @middleware
+    async def http_error_handler(req: Request, handler: Callable) -> Response:
+        """Handle HTTP errors.
+
+        :param req: A request instance
+        :param handler: A request handler
+        :returns: JSON response for the error
+        """
+        try:
+            response = await handler(req)
+            if response.status == 404:
+                return _json_exception(response.status,
+                                       Exception(response.message))
+            return response
+        except HTTPError as error:
+            return _json_exception(error.status, error)
+
+        # If something else goes wrong, convert to Internal Server Error
+        except Exception as ex:
```

Comment:
Like we discussed yesterday, this might hide some bugs and lead to weird situations. If program raises exceptions instead of httpErrors, we should add try-except to where the exception was raised.